main.ino

```diff
@@ -1,110 +1,119 @@
+ *
+ * This code is to control a small 'Swarmbot' consisting of DC motors,
+ * photosensors, hall effect sensors, collision detection switches, and
+ * a 2 MHz radio communication system.
+ *
+ * Notes
+ *     The bot's features are divided up into classes. Some classes encompass
+ *     others, such as driveControl containing two instances of motorControl.
+ *
+ *     Each class has three basic elements: a loop() function that executes
+ *     repeatedly, state variables that determine what loop() is doing, and
+ *     public functions to modify the state.
+ *
+ *     The main loop() function is simply here to call the loop() functions
+ *     of each of the bot's modules. Various test functions have also been
+ *     written to verify correct behavior.
+ *
+ *     The various loop() functions are all kept synchronized by a global
+ *     timer MILLIS, which is updated once per main loop(). That way each
+ *     loop() is guaranteed to execute in the correct sequence. Without a
+ *     global timer, it is possible for the millis() function to roll
+ *     over before the end of the main loop() is reached.
+ *
+ * Known issues
+ *     For some reason, the timing seems to be much more consistent when
+ *     doing micros() / 1000, rather than millis(). The downside is that
+ *     micros() rolls over after about 70 minutes, whereas micros() rolls
+ *     over after about 50 days. The risk of timer overflow is ignored here,
+ *     as the bot is not expected to perform for more than a few minutes at
+ *     a time.
+ *
+ *     Because of the structure outlined in the notes above, the CPU is
+ *     constantly crunching as fast as it can. The number of things that have
+ *     to run on each loop() is quite high, and this limits the maximum
+ *     frequency of a given function. For example, at the time of writing,
+ *     the LEDs in the lightSensor class have a minimum period of about 50 ms.
+ *
+ *     There's also nothing protecting MILLIS from being modified erroneously,
+ *     but given its unique purpose in the program, this is not so heinous.
+ */
+
+#include <Arduino.h>
+#include "globalTimer.h"
+
+extern uint32_t MILLIS;
+extern const int selectionPin;
+
+/* For each test you include, add the corresponding test from loop() */
+#include "testMotor.h"
+#include "testDrive.h"
+#include "testPathFollow.h"
+#include "testThermistor.h"
+#include "testBrightSensor.h"
+
+#include "testGoBeyond.h"
+#include "goBeyond.h"
+
+void (*scriptFunc)(void) = NULL;
+
+/* High baud rate -> fast printing, to minimize timing impact */
+void setup() {
+    Serial.begin(115200);
+
+    int script = analogRead(selectionPin) / 100;
+    
+    Serial.print("Executing Script: ");
+    
+    switch (script) {
+        case 10:
+            Serial.println("testGoBeyond"); 
+            goBeyond connection; 
+            connection.setup(); 
+            scriptFunc = testGoBeyond; 
+            break;

```

Comment:
I totally see what you were doing, but this is isn't how the jumper selector works. Each voltage level requires two cases to account for rounding error. (Frankly I'm not super happy with this system and in the branch I'm working on I've just commented it out.) 

If you still want to use the jumper selector, you should replace the "NO TEST" case down below instead of inserting your test here.

If this comment is confusing, I'd say just comment out the selector stuff and just call your function directly. The added complexity I was preparing for by adding this didn't end up being an issue.